src/Pay/Adapter/Stripe.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Utopia\Pay\Adapter;
+
+use Utopia\Pay\Adapter;
+
+class Stripe extends Adapter {
+
+    private string $baseUrl = 'https://api.stripe.com/v1';
+    private string $secretKey;
+    private string $publishableKey;
+
+
+    public function __construct(string $publishableKey, string $secretKey, string $currency = 'USD')
+    {
+        $this->secretKey = $secretKey;
+        $this->publishableKey = $publishableKey;
+        $this->currency = $currency;
+    }
+
+    /**
+     * Get name of the payment gateway
+     */
+    public function getName(): string {
+        return 'Stripe';
+    }
+  
+    /**
+     * Make a purchase request
+     */
+    public function purchase(int $amount, string $customerId, string $cardId = null, array $additonalParams = []): array {
+        $requestBody = [
+            'customer' => $customerId,
+            'amount' => $amount,
+            'currency' => $this->currency,
+        ];
+        $requestBody = array_merge($requestBody, $additonalParams);
+        if(!empty($cardId)) {
+            $requestBody['source'] = $cardId;
+        }
+        $res = $this->execute('POST', '/charges', $requestBody);
+        return $res;
+    }
+  
+    /**
+     * Refund payment
+     */
+    public function refund(string $paymentId, int $amount = null) : array {
+        $path = '/refunds';
+        $requestBody = ['charge' => $paymentId];
+        if($amount != null) {
+            $requestBody['amount'] = $amount;
+        }
+        return $this->execute('POST', $path, $requestBody);
+    }
+
+    /**
+     * Add a credit card for customer
+     */
+    public function createCard(string $customerId, string $cardId): array {
+        $path = '/customers/' . $customerId . '/sources';
+        return $this->execute('POST', $path, ['source' => $cardId]);
```

Comment:
This comment is true for all the methods. We should not return the entire response array to the calling code.
This array contains both the `responseBody` and the `headers`. The calling code does not need to know the response headers. 

We should parse the headers in this function or the `call` function in the Adapter throwing an error if status >= 400 and just pass on the `responseBody` otherwise

Reply:
The response headers are already discarded by the execute function in the Strip adapter. I added a separate execute method instead of calling the `call` method, as stripe needs to add user/password as curloption. Check the execute function.
Also I've modified slightly now so that `call` method throws exception on status > 400